forms/OptionList/index.js

```diff
@@ -43,10 +39,21 @@ export default React.createClass({
     }
   },
 
-  componentDidMount () {
-    let selectedAndCandidateRefs = this.getSelectedAndCandidateRefs()
-    let scrollPos = selectedAndCandidateRefs.selected.option.getDOMNode().offsetTop
-    this.refs.scrollContainer.getDOMNode().scrollTop = scrollPos
+  componentWillReceiveProps (nextProps) {
+    let { selectedOption } = nextProps
+    var newState = { focused: false }
+    if (selectedOption) {
+      newState = {
```

Comment:
Why do we have to set state this way. Should the prop always define the selection. On select the selection gets passed to the parent and the parent passes it down.This way you don't have to duplicate props onto state and there is a single source of truth.